### PR TITLE
add `dags.gitSync.extraEnv` list

### PR DESCRIPTION
**Pull request by n-oden**
**What does your PR do?**

Allow passing arbitrary env vars to the gitSync sidecar container.

## Checklist

- [x] Commits are [signed](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#sign-your-work)
- [x] Commits are [squashed](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#squash-commits) (if appropriate)
- [x] Chart [version bumped](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#versioning)
- [x] Chart [documentation updated](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#documentation)
- [x] Passes [linting](https://github.com/airflow-helm/charts/tree/main/CONTRIBUTING.md#linting)

### CODE REVIEW

**Comment by justincmoy** [comment]
I like this PR over https://github.com/airflow-helm/charts/pull/100.

**Comment by Yinabled** [comment]
Curious - would this be a means of addressing the case presented in #106? (i.e. adding the DAG directory to `PYTHONPATH` so as to include custom modules when syncing DAGs via `git-sync`)

**Comment by justincmoy** [comment]
> Curious - would this be a means of addressing the case presented in #106? (i.e. adding the DAG directory to `PYTHONPATH` so as to include custom modules when syncing DAGs via `git-sync`)

It didn't for me. I still needed to add the following to the top of our DAGs:
```
import importlib
import site
importlib.reload(site)
importlib.invalidate_caches()
```

**Comment by n-oden** [comment]
unless I'm missing something, changing PYTHONPATH in the context of the git-sync container won't have any effect on how DAGs are loaded at all -- the only thing running in the git-sync container is https://github.com/kubernetes/git-sync, which is a golang tool for synchronizing git repositories.

**Comment by justincmoy** [comment]
Sorry @n-oden, we're talking about the PYTHONPATH in airflow containers, like the web server and scheduler. I'm going to move this conversation to https://github.com/airflow-helm/charts/issues/106.

**Comment by n-oden** [comment]
@thesuperzapper thoughts? :)

File: charts/airflow/templates/_helpers/pods.tpl [hunk 1]
```diff
@@ -142,6 +142,9 @@ EXAMPLE USAGE: {{ include "airflow.container.git_sync" (dict "Values" .Values "s
           name: {{ .Values.dags.gitSync.httpSecret }}
           key: {{ .Values.dags.gitSync.httpSecretPasswordKey }}
     {{- end }}
+    {{- if .Values.dags.gitSync.extraEnv }}
+    {{- toYaml .Values.dags.gitSync.extraEnv | nindent 6 }}
```
Comment (justincmoy) [inline charts/airflow/templates/_helpers/pods.tpl:146]: Can you verify this works? I found I needed to change the nindent to `4`.

Comment (n-oden) [reply charts/airflow/templates/_helpers/pods.tpl:146]: oops, no, you're definitely correct -- this should be 4

**Comment by n-oden** [comment]
@justincmoy thoughts?

**Comment by justincmoy** [comment]
> @justincmoy thoughts?

I like it, but I'm not a contributor, so I have no powers on approving or merging. If I were them, I would ask you to bump the minor semver, so `8.1.0`, but I'll leave that up to them.

**Comment by thesuperzapper** [comment]
@n-oden why do you want to do this?

What does it let you do that you cant right now?

**Comment by n-oden** [comment]
@thesuperzapper I mean pretty much what it says on the tin: the k8s git-sync tool has a large number of options that are not currently accessible via this chart.

The specific issue that led me here was getting paged due to a transient network outage between our airflow cluster and github: we have a standing monitor for pods in CrashLoopBackoff for more than 10 minutes that it tripped.  This led to me opening https://github.com/airflow-helm/charts/pull/92, but the current chart had migrated from the homebrew git sync script to the kubernetes tool. Being able to set `GIT_SYNC_MAX_SYNC_FAILURES` is my real goal here but given the extensive configurability of the tool I figured it would be simpler and more future-proof to just make a general method of passing all possible env vars in.

**Comment by thesuperzapper** [comment]
@n-oden I am strongly against exposing extraEnv on the git-sync container, as it will confuse users, and could lead to them doing unexpected things.

Its much easier for us to just implement a new value like `dags.gitSync.maxFailures`, as described in issue https://github.com/airflow-helm/charts/issues/157. I will put in the next minor release.

**Comment by n-oden** [comment]
@thesuperzapper I can't say I agree: github.com/kubernetes/git-sync is well-established and well-documented, and one can't know a priori which configuration options any user of this chart might need.  If git-sync were an external chart installed transitively by this one (like redis is), all of that chart's options would be available even though our values.yaml only flags the options an airflow user is likely to care about; this seems like a equivalent pattern?

But hey, it's your project: I'll try to put up a PR for https://github.com/airflow-helm/charts/issues/157 this week.

**Pull request closed without merging**